File: JavaScript/프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    let stack = []
    
    for(let i = 0 ; i < s.length ; i++){
        if(!stack.length){
            stack.push(s[i])
            continue
        }
        
        let last = stack[stack.length-1]
        if(last == s[i]) stack.pop()
        else stack.push(s[i])
    }
    
    if(!stack.length) return 1
    return 0
}

File: Python/프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stack = []
    
    for i in range(len(s)) :
        if not len(stack) : 
            stack.append(s[i])
            continue
        
        last = stack[-1]
        if last == s[i] : stack.pop()
        else : stack.append(s[i])
    
    
    if not len(stack) : return 1
    return 0